dpnp/backend/extensions/ufunc/elementwise_functions/interpolate.cpp

```diff
@@ -0,0 +1,296 @@
+    const T *right = static_cast<const T *>(vright);
+    T *out = static_cast<T *>(vout);
+
+    using dpnp::kernels::interpolate::interpolate_impl;
+    sycl::event interpolate_ev = interpolate_impl<TCoord, T>(
+        exec_q, x, idx, xp, fp, left, right, out, n, xp_size, depends);
+
+    return interpolate_ev;
+}
+
+interpolate_fn_ptr_t interpolate_dispatch_vector[td_ns::num_types];
+
+void common_interpolate_checks(
+    const dpctl::tensor::usm_ndarray &x,
+    const dpctl::tensor::usm_ndarray &idx,
+    const dpctl::tensor::usm_ndarray &xp,
+    const dpctl::tensor::usm_ndarray &fp,
+    const dpctl::tensor::usm_ndarray &out,
+    const std::optional<const dpctl::tensor::usm_ndarray> &left,
+    const std::optional<const dpctl::tensor::usm_ndarray> &right)
+{
+    array_names names = {{&x, "x"}, {&xp, "xp"}, {&fp, "fp"}, {&out, "out"}};
+
+    auto array_types = td_ns::usm_ndarray_types();
+    int x_type_id = array_types.typenum_to_lookup_id(x.get_typenum());
+    int idx_type_id = array_types.typenum_to_lookup_id(idx.get_typenum());
+    int xp_type_id = array_types.typenum_to_lookup_id(xp.get_typenum());
+    int fp_type_id = array_types.typenum_to_lookup_id(fp.get_typenum());
+    int out_type_id = array_types.typenum_to_lookup_id(out.get_typenum());
+
+    if (x_type_id != xp_type_id) {
+        throw py::value_error("x and xp must have the same dtype");
+    }
+    if (fp_type_id != out_type_id) {
+        throw py::value_error("fp and out must have the same dtype");
+    }
+    if (idx_type_id != static_cast<int>(td_ns::typenum_t::INT64)) {
+        throw py::value_error("The type of idx must be int64");
+    }
+
+    auto left_v = left ? &left.value() : nullptr;
+    if (left_v) {
+        names.insert({left_v, "left"});
+        if (left_v->get_ndim() != 0) {
+            throw py::value_error("left must be a zero-dimensional array");
+        }
+
+        int left_type_id =
+            array_types.typenum_to_lookup_id(left_v->get_typenum());
+        if (left_type_id != fp_type_id) {
+            throw py::value_error(
+                "left must have the same dtype as fp and out");
+        }
+    }
+
+    auto right_v = right ? &right.value() : nullptr;
+    if (right_v) {
+        names.insert({right_v, "right"});
+        if (right_v->get_ndim() != 0) {
+            throw py::value_error("right must be a zero-dimensional array");
+        }
+
+        int right_type_id =
+            array_types.typenum_to_lookup_id(right_v->get_typenum());
+        if (right_type_id != fp_type_id) {
+            throw py::value_error(
+                "right must have the same dtype as fp and out");
+        }
+    }
+
+    common_checks({&x, &xp, &fp, left_v, right_v}, {&out}, names);
+
+    if (x.get_ndim() != 1 || xp.get_ndim() != 1 || fp.get_ndim() != 1 ||
+        idx.get_ndim() != 1 || out.get_ndim() != 1)
+    {
+        throw py::value_error("All arrays must be one-dimensional");
+    }
+
+    if (xp.get_size() != fp.get_size()) {
```

Comment:
Add function `check_same_size` and/or `check_same_shape`?

Reply:
done, thank you